### PR TITLE
Docker file

### PR DESCRIPTION
**Pull request by nialexsan**
Closes: #???

## Description

Dockerfile for a build to deploy

______

For contributor use:

- [x] Targeted PR against `master` branch
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [standards mentioned here](https://github.com/onflow/flow-nft/blob/master/CONTRIBUTING.md#styleguides).
- [ ] Updated relevant documentation 
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Added appropriate labels

### CODE REVIEW

**Review by franklywatson — APPROVED** [review]

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -2,8 +2,8 @@ name: "Deploy EVM Gateway to Cloud Run"
 
 on:
   push:
-    branches:
-      - main
+    # branches:
```
Comment (m-Peter) [inline .github/workflows/deploy.yml:5]: Is this intentional? Do we want them to be commented-out?

File: api/server_test.go [hunk 2]
```diff
@@ -45,7 +45,7 @@ func TestServerJSONRPCOveHTTPHandler(t *testing.T) {
 	blockchainAPI := api.NewBlockChainAPI(config, store, mockFlowClient)
 	supportedAPIs := api.SupportedAPIs(blockchainAPI)
 	srv.EnableRPC(supportedAPIs)
-	srv.SetListenAddr("localhost", 8545)
+	srv.SetListenAddr("", 8545)
```
Comment (m-Peter) [inline api/server_test.go:48]: I'm curious, did you encounter a problem with `"localhost"`?

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+# Install go modules
+WORKDIR /flow-evm-gateway
+COPY go.* ./
+COPY . ./
+RUN go mod download
+RUN go mod verify
+RUN CGO_ENABLED=0 go build -o evm-gateway ./cmd/server/main.go
+RUN chmod a+x evm-gateway
+RUN chmod a+x ./scripts/run.sh
+RUN git clone https://github.com/m-Peter/flow-cli-custom-builds.git
```
Comment (m-Peter) [inline Dockerfile:14]: Minor, I run into a glitch the first time building the image locally:
```bash
> [builder 10/10] RUN git clone https://github.com/m-Peter/flow-cli-custom-builds.git:
0.344 Cloning into 'flow-cli-custom-builds'...
66.27 error: 3872 bytes of body are still expected
66.28 fetch-pack: unexpected disconnect while reading sideband packet
66.29 fatal: early EOF
66.31 fatal: fetch-pack: invalid index-pack output
------
ERROR: failed to solve: executor failed running [/bin/sh -c git clone https://github.com/m-Peter/flow-cli-custom-builds.git]: exit code: 128
```

Maybe another alternative could be:
```Dockerfile
RUN wget https://github.com/m-Peter/flow-cli-custom-builds/raw/main/flow-x86_64-linux-
RUN chmod a+x flow-x86_64-linux-
```

With the 2nd attempt, I was able to build the image and run a container, so it is not a big deal.

**Review by m-Peter — APPROVED** [review]
LGTM! :rocket: 
I built an image locally and ran the container. It worked very well :clap:

File: .github/workflows/deploy.yml (hunk 1 above)
Comment (nialexsan) [reply .github/workflows/deploy.yml:5]: I'll revert these changes, they were only for testing

File: api/server_test.go (hunk 2 above)
Comment (nialexsan) [reply api/server_test.go:48]: yes, there was an issue in cloud run
it's a specific requirement for cloud run 
```
The ingress container within an instance must listen for requests on 0.0.0.0 on the port to which requests are sent.
```
https://cloud.google.com/run/docs/container-contract

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+# RUN APP
+FROM debian:latest
+WORKDIR /flow-evm-gateway
+COPY --from=builder /flow-evm-gateway/evm-gateway /flow-evm-gateway/evm-gateway
+COPY --from=builder /flow-evm-gateway/flow-cli-custom-builds/flow-x86_64-linux- /flow-evm-gateway/flow-x86_64-linux-
+COPY --from=builder /flow-evm-gateway/flow.json /flow-evm-gateway/flow.json 
+COPY --from=builder /flow-evm-gateway/api/cadence/transactions/create_bridged_account.cdc /flow-evm-gateway/create_bridged_account.cdc 
+COPY --from=builder /flow-evm-gateway/scripts/run.sh /flow-evm-gateway/run.sh
+EXPOSE 8545
```
Comment (franklywatson) [inline Dockerfile:26]: We'll need to expose this so it can be templated and set externally

File: api/server_test.go (hunk 2 above)
Comment (franklywatson) [reply api/server_test.go:48]: Similar comment as above about the port needing to be an injected variable

File: api/server_test.go [hunk 5]
```diff
@@ -460,7 +460,7 @@ func TestServerJSONRPCOveWebSocketHandler(t *testing.T) {
 	blockchainAPI := api.NewBlockChainAPI(config, store, flowClient)
 	supportedAPIs := api.SupportedAPIs(blockchainAPI)
 	srv.EnableWS(supportedAPIs)
-	srv.SetListenAddr("localhost", 8545)
+	srv.SetListenAddr("", 8545)
```
Comment (franklywatson) [inline api/server_test.go:463]: Same comment

File: cmd/server/main.go [hunk 6]
```diff
@@ -182,7 +182,7 @@ func runServer(config *api.Config, store *storage.Store, logger zerolog.Logger)
 	srv.EnableRPC(supportedAPIs)
 	srv.EnableWS(supportedAPIs)
 
-	srv.SetListenAddr("localhost", 8545)
+	srv.SetListenAddr("", 8545)
```
Comment (franklywatson) [inline cmd/server/main.go:185]: Same again

File: scripts/run.sh [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+while ! grep -q "$PATTERN" temp_output.txt; do
+  sleep 1
+done
+
+# Once the pattern is found, you can kill the first process if needed
+# kill $FIRST_PROCESS_PID
+
+# Run the second process
+
+./flow-x86_64-linux- transactions send /flow-evm-gateway/create_bridged_account.cdc 1500.0 --network=emulator --signer=emulator-account && ./evm-gateway
```
Comment (franklywatson) [inline scripts/run.sh:20]: We will need to be able to pass in the network to be used, whether emulator or other network

**Pull request merged**